### PR TITLE
ARTEMIS-1523 Allow MQTT with dynamic cluster

### PR DESCRIPTION
**Pull request by Skiler**
ARTEMIS-1523 Allow MQTT with dynamic cluster

### CODE REVIEW

**Comment by Skiler** [comment]
Hi @jbertram 

This is the new pull request. It's possible to include this fix in the 2.4.0 version?
Thanks

File: artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTPublishManager.java [hunk 1]
```diff
@@ -5,9 +5,9 @@
  * The ASF licenses this file to You under the Apache License, Version 2.0
  * (the "License"); you may not use this file except in compliance with
  * the License. You may obtain a copy of the License at
- *
+ * <p>
  * http://www.apache.org/licenses/LICENSE-2.0
- *
+ * <p>
```
Comment (michaelandrepearce) [inline artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTPublishManager.java]: <p> should not be added

Comment (michaelandrepearce) [reply artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTPublishManager.java]: On that note, there doesn't seem to be any material change to this class, except you've used a re-formatter set to your own settings, id suggest simply reverting this file.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/WildcardAddressManager.java [hunk 2]
```diff
@@ -30,6 +25,11 @@
 import org.apache.activemq.artemis.core.postoffice.BindingsFactory;
 import org.apache.activemq.artemis.core.transaction.Transaction;
 
+import java.util.Collection;
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/WildcardAddressManager.java]: This move re-change wasn't needed, if could revert, so just the code changes needed are in the PR.

**Comment by jbertram** [comment]
@michaelandrepearce, I'm working on this PR with @Skiler.  I plan on amending it before merging to deal with all the stuff you mentioned (plus a few other things).

**Comment by jbertram** [comment]
@Skiler, btw this will go into 2.5.0 since 2.4.0 has already been released.

**Comment by michaelandrepearce** [comment]
@jbertram great cheers for that.

**Pull request closed without merging**